server/stycoon/reports.go

```diff
@@ -0,0 +1,219 @@
+package stycoon
+
+import (
+	"database/sql"
+	"strconv"
+
+	"github.com/rs/zerolog/log"
+)
+
+// nil tick means fetching all previous ticks
+func (game *Game) fillCombats(tick *int64) error {
+	var rows *sql.Rows
+	var err error
+	if tick == nil {
+		rows, err = game.db.Query("select `tick`, `attacker`, `defender`, `killed` from t_report_combat order by `tick`")
+	} else {
+		rows, err = game.db.Query("select `attacker`, `defender`, `killed` from t_report_combat where tick = ?", *tick)
+	}
+	if err != nil {
+		return err
+	}
+	var combat Combat
+	var killed string
+	for rows.Next() {
+		if tick == nil {
+			rows.Scan(&combat.Tick, &combat.Attacker, &combat.Defender, &killed)
```

Comment:
this is repeatedly changing the same referenced objects therefore the array in the end will have multiple times the same data

Reply:
done